### DISCUSSION TITLE
Change verbosity for tessellation

### DISCUSSION THREAD

**Discussion by DorianDepriester** [neperfepx/neper#241] [Q&A] [answered]
Hello there!
I would like to turn off real-time optimization info. Indeed, when running on remote computer (e.g. on HPC cluster), one usually dumps the output message to a text file. On large tessellation, this file can be enormous because of repeated lines such as

    Iteration  32870: fmin=0.193153552 f=0.193280460

So my question is quite simple: is there a way to change verbosity for tessellation module? 

Regards.

**Comment by rquey** [comment, marked as answer]
Are you redirecting Neper's terminal output to a file, using `>`? You could use `grep -v Iteration` to strip out the iteration lines:
```
$ neper -T ... | grep -v Iteration > neper.log
```
If redirection is done "automatically", you can still use `neper -T ... | grep -v Iteration`...

**Reply by DorianDepriester** [reply]
Not exactly. Usually I use ``screen`` on remote computers and it works well. My concern here is on a HPC cluster running on the SLURM job scheduler. In this case, the stdout of the running job is redirected to a given log file. 
So basically I think it is the same as ``>``.

**Reply by rquey** [reply]
So, `neper -T ... | grep -v Iteration` should work...

**Reply by DorianDepriester** [reply]
My bad, I hadn't seen your edit when I answered. I will try to append the grep stuff in the submitted command.
Thank you @rquey for your quick answer!

**Reply by DorianDepriester** [reply]
> So, `neper -T ... | grep -v Iteration` should work...

As an improvement:

    neper -T ... | grep --line-buffered -v Iteration

The ``--line-buffered`` will enable continuous stream (instead of waiting for grep to flush)

**Reply by yguilhem** [reply]
Thank you for the `--line-buffered` hint!
However, it does not allow to keep track of the total number of iterations needed for a specific tesselation process.
@rquey: Would it be possible to add a "final solution line" to the output (as it is for the initial solution) that would look like the following?
```
Final solution: f=0.002392548 (iteration 111909)
```
This way, we could avoid all the solution history in the log but keep track of initial and final steps.

**Reply by rquey** [reply]
This will be done in the upcoming release.

**Reply by yguilhem** [reply]
@rquey: Thank you for the new output in the very new release. Sorry to insist about this, but would it be possible to also add the final number of iterations to the final solution output line(s)?

**Reply by rquey** [reply]
Done in `4.2.1-1`:
```
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.2.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 8 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2021, and GNU GPL'd, by Romain Quey.
Info   : Loading initialization file `/home/rquey/.neperrc'...
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] -tesrformat ascii
Info   : [com line] -n 2 -morpho gg
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :   - Setting seeds... 100%
Info   :   - Generating crystal orientations...
Info   :   - Running tessellation...
Info   :     > Initial solution: f   =13.325428096
Info   :     > Iteration    307: fmin=1.800334442 f=1.800334577  
Info   :     > Final solution  : f   =1.800334442 (307 iterations)
Info   :     > Reached `eps' criterion.
Info   : Writing results...
Info   :     [o] Writing file `n2-id1.tess'...
Info   :     [o] Wrote file `n2-id1.tess'.
Info   : Elapsed time: 0.278 secs.
========================================================================

```
